service/history/shard/controller_impl.go

```diff
@@ -344,8 +361,10 @@ func (c *ControllerImpl) acquireShards() {
 						c.contextTaggedLogger.Error("Error looking up host for shardID", tag.Error(err), tag.OperationFailed, tag.ShardID(shardID))
 					} else {
 						if info.Identity() == c.hostInfoProvider.HostInfo().Identity() {
+							// Wait up to 1s for the shard to acquire the rangeid lock.
+							// After 1s we will move on but the shard will continue trying in the background.
```

Comment:
GetShardByID internally discard ctx, so it does not honer that 1s timeout. Can you fix that as well?

Reply:
This does fix it, that's the whole point of the PR :)

Reply:
In the new version, GetShardByID no longer takes a ctx. It doesn't do any persistence operations, so it doesn't make sense to take a ctx.. there's nothing cancelable.